src/OpenLoco/GameCommands/ChangeCompanyFace.cpp

```diff
@@ -0,0 +1,87 @@
+#include "../CompanyManager.h"
+#include "../Interop/Interop.hpp"
+#include "../Localisation/StringIds.h"
+#include "../Objects/CompetitorObject.h"
+#include "../Objects/ObjectManager.h"
+#include "../Ui/WindowManager.h"
+#include "GameCommands.h"
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x00435506
+    static uint32_t changeCompanyFace(uint8_t flags, CompanyId targetCompanyId, ObjectHeader& targetHeader)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Miscellaneous);
+
+        // Find target competitor object among loaded objects
+        auto foundCompetitor = ObjectManager::findObjectHandle(targetHeader);
+
+        // See whether any other company is using this competitor already
+        for (auto& company : CompanyManager::companies())
```

Comment:
This find should be behind a `if (foundCompetitor.has_value())`.
```cpp
auto& companies = CompanyManager::companies();
auto res = std::find_if(std::begin(companies), std::end(companies),
    [id = foundCompetitor->id](auto& company){
        return company.competitorId == id;
    });
if (res != std::end(companies)){
    ...
}
```

Reply:
I'm not sure using `std::find_if` would improve the readability of the code block tbh. I have wrapped the block in a `if (foundCompetitor)` for now.